ReactTask/src/modules/Redux.modules.ts

```diff
@@ -0,0 +1,74 @@
+export interface AuthReducerType {
+  user?: {
+    id: string;
+    first_name: string;
+    last_name: string;
+    photo: string;
+    role_name: string;
+  };
+  token?: string;
+  error?: string;
+  isAuthenticated: boolean;
+}
+
+export interface IAddAppointmentResponse {
+  id: string;
+  patient_id: string;
+  doctor_id: string;
+  visit_date: string;
+  reason: string;
+  note: string;
+  status: string;
+}
+
+export interface IAddAppointment {
+  date: string;
+  reason: string;
+  note: string;
+  doctorID: string;
+}
+
+export interface IAddAppointmentResponse {
+  id: string;
+  patient_id: string;
+  doctor_id: string;
+  visit_date: string;
+  reason: string;
+  note: string;
+  status: string;
+}
+
+export interface ILogInForm {
+  userName: string;
+  password: string;
+}
+
+export interface IRegisterForm {
+  userName: string;
+  password: string;
+  firstName: string;
+  lastName: string;
+}
+
+export interface ITokens {
+  access_token: string;
+  refresh_token: string;
+}
+
+export interface IProfileWithData {
+  data: {
+    id: string;
+    first_name: string;
+    last_name: string;
+    photo: string;
+    role_name: string;
+  };
```

Comment:
By the way now i understand that in my code I can do the same! Make distructure in your file with request. Instead of response = ... rewrite {data} = ... With help of these movements u can avoid code dublicates and have only 1 interface

Reply:
Sorry. I noticed that we can't do this because of having another data variable in scope <--Cannot redeclare block-scoped variable 'data'.-->

Reply:
I'm sorry, but I didn't understand you :(